optuna/storages/_redis.py

```diff
@@ -505,6 +524,18 @@ def _set_best_trial(self, study_id: int, trial_id: int) -> None:
             pipe.set(self._key_study_summary(study_id), pickle.dumps(study_summary))
             pipe.execute()
 
+    def _check_and_set_param_distribution(
```

Comment:
This method is used in `_CachedStorage`. I think we should define this method in the `BaseStorage` for the maintainability. What do you think?

Reply:
I do not think so...
Because ```BaseStorage``` is also inhedited by ```InMemoryStorage``` and it does not use ```_check_and_set_param_distribution```.
```
BaseStorage
 |
 | inherit
 |
 +-- _CachedStorage (calls _check_and_set_param_distribution of wrapping class)
 |    |
 |    | wrap
 |    |
 |    +--- RDBStorage
 |    |
 |    +--- RedisStorage
 |
 +-- InMemoryStorage (does not calls _check_and_set_param_distribution because InMemoryStorage does not wrapping class)
```

Reply:
Even though it will not be used by `InMemoryStorage`, we can define it in `BaseStorage`. Also, it will be better if we will define another wrapped storeges.
Could you move the definition to `BaseStorage`?

Reply:
That fix seems to be beyond the scope of this PR... So I will fix it in another PR after this PR is merged. Is it OK ?